### PR TITLE
Fixed patch for VF fix, added profiling support

### LINKED ISSUE #52: Broken SVML autovectorization for some of math functions
This is pretty much my bug introduced in #47 due to insufficient unit-testing (I need `llvm-lit` from llvm-utils be build in order to run unit tests).

Issue:

Testing llvmdev=7 with Numba, revealed that VF (vectorization factor) fix was not enough or there is another issue of integration with SVML patch:
```
$ conda create -n cf-numba -c conda-forge numba llvmdev=7
$ source activate cf-numba
$ conda install -c numba --no-deps icc_rt
$ conda list | egrep 'llvm|numba'
# packages in environment at miniconda3/envs/cf-numba:
icc_rt                    2018.0.2                      0    numba
llvm-meta                 7.0.0                         0    conda-forge
llvmdev                   7.0.0                h2d50403_0    conda-forge
llvmlite                  0.25.0           py36hd28b015_1    conda-forge
numba                     0.40.0           py36hf8a1672_0    conda-forge

$ NUMBA_DUMP_LLVM=1 python -m numba.runtests numba.tests.test_svml.TestSVMLGeneration.test_int32_prange4_usecase | egrep -v '===|---' >llvm7_issue.ll

$ opt -S -vector-library=SVML -mcpu=haswell -O3 llvm7_issue.ll | grep exp
  %.397 = tail call double @expm1(double %.396)
  %200 = call intel_svmlcc <8 x double> @__svml_exp8_ha(<8 x double> %199)
  %.3487 = tail call double @llvm.exp.f64(double %.3486)
  %411 = call intel_svmlcc <4 x double> @__svml_expm14_ha(<4 x double> %shuffle)
  %412 = call intel_svmlcc <4 x double> @__svml_expm14_ha(<4 x double> %shuffle1788)
declare double @expm1(double) local_unnamed_addr #1
declare double @llvm.exp.f64(double) #2
declare <4 x double> @__svml_expm14_ha(<4 x double>) local_unnamed_addr #1
declare <8 x double> @__svml_exp8_ha(<8 x double>) #2
```
Please note the mixed usage of 4x and 8x vector lengths. The latter is wrong (or just `grep 8_ha`)

### PR DESCRIPTION
**Pull request by anton-malakhov**
Fixes #52 
Also enables profiler support (opt-in, depends on enabling for ExecutionEngine)

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

So far, I have not found how to enable regression testing on Windows (except for my numba test).. but it takes 80 minutes to build it now, so anyway, there is not enough room for testing.

### CODE REVIEW

**Comment by anton-malakhov** [comment]
@conda-forge-admin, please rerender

**Comment by anton-malakhov** [comment]
I found no better way of enabling `llvm-lit` for testing but using this hand-crafted `partial-testing.patch`. The best what we can do without it is running the `numba-3016.ll` regression test only but it can lead to other issues in future. If you want to run it on the conda's test stage, then `LLVM_INSTALL_UTILS` should be enabled, which will include additional files into the package (`FileCheck` and others).
So, I think the current state of the patch is a good trade-off for sake of enabling as much unit-testing as possible.
Please review/merge, @jakirkham @mingwandroid @isuruf

**Comment by anton-malakhov** [comment]
I'd appreciate if anybody leaves any comment here..

File: recipe/bld.bat [hunk 1]
```diff
@@ -8,15 +8,22 @@ cmake -G "Ninja" ^
...
 
 if errorlevel 1 exit 1
 
 cmake --build . --target install
 if errorlevel 1 exit 1
+
+bin\opt -S -vector-library=SVML -mcpu=haswell -O3 %RECIPE_DIR%\numba-3016.ll | bin\FileCheck %RECIPE_DIR%\numba-3016.ll
+if errorlevel 1 exit 1
+
+rem cd ..\test
```
[the remark this answers is not in the document]
Comment (anton-malakhov) [reply recipe/bld.bat]: well.. Ideally, I'd like to fix windows regression testing as well. But.
1. I don't know why Windows is different for this cmake configuration, which works for linux and mac. It does not produce the same working `llvm-lit` tool. Thus I dunno how to run it. 
2. it almost hits Appveyor time limits anyway, thus there is no much room to run these tests anyway.
Thus I commented it out intentionally and you basically can just remove these lines completely or add a TODO comment.

**Comment by jakirkham** [comment]
Are these patches still reasonable on top of LLVM 7.0.1? Have merged the latest changes from `master` into this PR to try building with them, but it would be good to know if that works for you as well.

**Comment by jakirkham** [comment]
No worries. Figured they hadn't landed in the patch release.

Cool, would be great to have them in LLVM 8. Guessing they have ~1yr to get that sorted. ;)

That would be great if you have time to take a look. There are new llvmlite and Numba releases coming out that we would like to get built soon. So it would be nice to roll these changes into those builds as well. :)

**Comment by stuartarchibald** [comment]
I seem to recall testing 7.0.1 when trying to work out what was causing this https://bugs.llvm.org/show_bug.cgi?id=39905 (unmerged patch needed), so think it has a fair chance of working as-is.

I fixed up the windows testing but ended up having to comment it as the `llvm-lit.py` test inputs often seem to use `grep` etc which are require a *nix shell. It may be possible to fix this but I've not investigated further. The code is here, but commented https://github.com/numba/llvmlite/blob/894a1ddab60091869424f697cc56ecfee8156e31/conda-recipes/llvmdev/bld.bat#L36-L40

Numba 0.42/llvmlite 0.27 is almost complete in terms of release building, packages are on the Numba channel already :)

**Pull request merged**